### PR TITLE
Update welcome page of the user docs.

### PR DESCRIPTION
**Pull request by deoxys94**
## Description

- Update the contents to align with the style guide (Remove passive voice and ambiguous pronouns, use a friendly, more conversational tone, shorten sentences to improve readability).
- Change the headers to sentence case.
- Restructure the contents using tables to improve scannability.
- Add short descriptions for some of the missing sections.

### Submitter Checklist

- [ ] Squashed commits with `git rebase -i` (if needed)

### CODE REVIEW

File: docs/user/intro.md [hunk 1]
```diff
@@ -6,51 +6,36 @@ sidebar_position: 0
...
+| [Contributing](/docs/category/contributing)          | Find how you can contribute to Solus, whether you're a beginner or an experienced Linux and Solus user.                                                                                     |
+| [Editions](/docs/user/editions/)                     | Understand the different editions of Solus: differences between editions, configuration tasks, and helpful tips and tricks.                                                                |
+| [Hardware](/docs/user/hardware/)                     | Solus supports most hardware out-of-the-box, but there are exceptions. This section focuses on those exceptions and how to get the most out of your hardware. |
+| [Package management](/docs/user/package-management/) | Learn how to install software and manage the repositories of your Solus system.                                                                                                             |
```
Comment (davidjharder) [inline docs/user/intro.md:21]: This gives me a 404 when I test locally. I think the trailing slash may need to be removed

File: docs/user/intro.md [hunk 2]
```diff
@@ -6,51 +6,36 @@ sidebar_position: 0
...
+| [Privacy policy](/docs/user/privacy.md)              | Understand our privacy policy: Learn how we handle your data, and how to contact us for privacy concerns.                                                                                   |
+| [Software](/docs/user/software/)                     | Getting started and configuration guides for the software packages included in Solus.                                                                                                       |
```
Comment (davidjharder) [inline docs/user/intro.md:23]: This gives me a 404 when I test locally. I think the trailing slash may need to be removed

**Review by davidjharder — CHANGES_REQUESTED** [review]
Thanks for the PR. I like this re-arrangement.
Here are some initial review comments, I'll review the writing a little later.

**Comment by davidjharder** [comment]
Writing looks good.

Please run `prettier` to format the tables.

**Comment by davidjharder** [comment]
This needs a rebase now.

File: docs/user/intro.md (hunk 2 above)
Comment (deoxys94) [reply docs/user/intro.md]: Fixed

**Comment by deoxys94** [comment]
Hi @davidjharder,

I fixed the links and rebased the branch.

By the way, why software and package management are under `docs/category/` instead of  `docs/user`? The arrangement seems a little unusual.

**Comment by davidjharder** [comment]
> By the way, why software and package management are under `docs/category/` instead of `docs/user`? The arrangement seems a little unusual.

Yeah, it's weird. I don't know docusaurus well enough to do anything other than go along with it for now.

**Review by davidjharder — APPROVED** [review]
Looks good. Thanks for your continued work on the help center

**Pull request merged**